pkg/capture/capture_manager.go

```diff
@@ -380,6 +381,48 @@ func (cm *Manager) update(ctx context.Context, ifaces config.Ifaces, enable, dis
 	rg.Wait()
 }
 
+// GetFlowMaps extracts a copy of all active flows and sends them on the provided channel (compatible with normal query
+// processing). This way, live data can be added to a query result
+func (cm *Manager) GetFlowMaps(ctx context.Context, filterFn goDB.FilterFn, writeoutChan chan<- hashmap.AggFlowMapWithMetadata, ifaces ...string) {
+
+	logger, t0 := logging.FromContext(ctx), time.Now()
+
+	// Build list of interfaces to process (either from all interfaces or from explicit list)
+	// If none are provided / are available, return empty map
+	if ifaces = cm.captures.Ifaces(ifaces...); len(ifaces) == 0 {
+		return
+	}
+
+	for _, iface := range ifaces {
+		mc, exists := cm.captures.Get(iface)
+		if exists {
+
+			runCtx := withIfaceContext(ctx, mc.iface)
+
+			// Lock the running capture and perform the rotation
+			mc.lock()
+			flowMap := mc.flowMap(runCtx)
+			mc.unlock()
+
+			if flowMap != nil {
+				if filterFn != nil {
+					flowMap = filterFn(flowMap)
+				}
+				writeoutChan <- hashmap.AggFlowMapWithMetadata{
+					AggFlowMap: flowMap,
+					Interface:  iface,
+				}
+			}
+		}
+	}
+
+	// log fetch duration
+	logger.With(
```

Comment:
Now that I see this: an additional metric for this would be cool.

Reply:
I thought so too for a second, but thing is: This will only ever be called if live data is used anywhere, so you might end up with a stupid metric that has low statistic anyways and hence doesn't tell you anything. IMHO I'd rather be interested in the time the _in-memory_ rotation takes (on top of the overall rotation time we currently already track), but as I hinted elsewhere (I think) that would mostly be relevant on a per-interface level (and hence might not be reasonable). I think having this debug statement to be able to figure out the overhead (which will be minuscule anyway) is enough...